### PR TITLE
Refactor policy state verification

### PR DESCRIPTION
**Pull request by adityasaky**
Since rule names are unique across all metadata in a policy state, we don't have to worry about diamond delegations. So, we can verify all signatures on policy metadata as well as check if some file is not reachable. With this change, FindVerifiersForPath no longer needs to verify delegated metadata signatures.

Separately, state.Verify is now limited to internal consistency checks. So, it verifies that root keys in the state match, followed by checking all the targets role signatures as discussed above.

Finally, LoadCurrentState now inspects all policy states to verify root role on each state using the previous one. The very first one is still TOFU, but this can be bootstrapped using other mechanisms.

Blocked by #304.

### CODE REVIEW

**Comment by adityasaky** [comment]
I want us to be really careful with this one and we might want to clarify in docs what's happening as we go through this. The review on this can wait until the first three commits land separately.

File: internal/policy/policy.go
Comment (adityasaky) [reply internal/policy/policy.go:421]: Added a comment. This is because the last rule is always the allow rule. We don't actually process it during our search, but it's there explicitly to indicate to the user how rules work.

**Review by wlynch — APPROVED** [review]

**Pull request merged**
